homeassistant/components/goalzero/const.py

```diff
@@ -42,112 +42,136 @@
 
 MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=30)
 
-BINARY_SENSOR_DICT = {
-    "backlight": {ATTR_NAME: "Backlight", ATTR_ICON: "mdi:clock-digital"},
-    "app_online": {
-        ATTR_NAME: "App Online",
-        ATTR_DEVICE_CLASS: DEVICE_CLASS_CONNECTIVITY,
-    },
-    "isCharging": {
-        ATTR_NAME: "Charging",
-        ATTR_DEVICE_CLASS: DEVICE_CLASS_BATTERY_CHARGING,
-    },
-    "inputDetected": {
-        ATTR_NAME: "Input Detected",
-        ATTR_DEVICE_CLASS: DEVICE_CLASS_POWER,
-    },
-}
+BINARY_SENSOR_TYPES: tuple[BinarySensorEntityDescription, ...] = (
```

Comment:
Side note: It's better to move each description container to its corresponding platform since these are details concerning only each platform.